### PR TITLE
OV python: update getting pytorch tensor as input

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - fix for pytorch tensors usage as input . Currently OV failed to infer with pytorch tensor under gradient calculation or offloaded on cuda device, due to inability to get numpy array from it. This fix prevent such behaviour.

### CODE REVIEW

File: src/bindings/python/src/openvino/runtime/ie_api.py [hunk 1]
```diff
@@ -101,6 +101,8 @@ def normalize_inputs(request: InferRequestBase, inputs: dict) -> dict:
             new_inputs[key] = value
         # If value object has __array__ attribute, load it to Tensor using np.array.
         elif hasattr(value, "__array__"):
+            if hasattr(value, "detach"):
+                value = value.detach().cpu()
```
Comment (rkazants) [inline src/bindings/python/src/openvino/runtime/ie_api.py:105]: may we have one test for this please?

Comment (eaidova) [reply src/bindings/python/src/openvino/runtime/ie_api.py:105]: sure, but I need to discuss with python api developers before, is it allowed to use torch in tests or not

Comment (eaidova) [reply src/bindings/python/src/openvino/runtime/ie_api.py:105]: BTW, @rkazants maybe you know, is there some tricky cases for tf tensors too?

Comment (rkazants) [reply src/bindings/python/src/openvino/runtime/ie_api.py:105]: Input and output tensors from TF1 can be treated as simple numpy arrays. As for TF2 there is no problem with input tensors but I see for handling output tensors we use `tf_res.numpy()` [here](https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/common/tf2_layer_test_class.py#L63)

Comment (rkazants) [reply src/bindings/python/src/openvino/runtime/ie_api.py:105]: I think there should not be a problem to use `torch` model only for tests now. We need to update `openvino\src\bindings\python\requirements_test.txt`

**Comment by p-wysocki** [comment]
This change would require including `torch` in `requiremets_test.txt`. It would be the only test requiring it, and the package wheel is relatively big (https://pypi.org/project/torch/#files).

I think the change is needed, but we need to discuss how we should go about testing it.

File: src/bindings/python/src/openvino/runtime/ie_api.py (hunk 1 above)
Comment (jiwaszki) [reply src/bindings/python/src/openvino/runtime/ie_api.py:105]: To be honest I think adding whole PyTorch dependency might be "too heavy" as we are adding only one test to runtime. Are there any other goals? I look at something like ONNX dependency that test many scenarios for ONNX Frontend.

Other than that, why `detach()` is going to be used? It will trigger for all PyTorch based tensors. Is it always a case? As far as I know Tensor created after call to `detach()` [is sharing memory](https://pytorch.org/docs/stable/generated/torch.Tensor.detach.html#torch.Tensor.detach):

> Returned Tensor shares the same storage with the original one. In-place modifications on either of them will be seen, and may trigger errors in correctness checks.

So in specific cases, users are required to secure/not modify original Tensors and to keep alive original Tensors - so memory is not cleared by Python's garbage collector before OV inference. I think this scenarios should be checked first and **I am strongly against "non-copy" approach when dealing with memory that is created outside OpenVINO control flow**. `openvino.runtime.Tensor` class is always copying memory and memory-sharing with numpy arrays is available after explicit flag setting. Please also look at the approach of PyTorch:
image

To sum things up, I do not think this change should include `detach()` without copying of memory. It might be dangerous and should be tested out.

**Comment by rkazants** [comment]
> This change would require including `torch` in `requiremets_test.txt`. It would be the only test requiring it, and the package wheel is relatively big (https://pypi.org/project/torch/#files).
> 
> I think the change is needed, but we need to discuss how we should go about testing it.

In any case earlier or later, we will have to add this dependency for PyTorch Frontend validation. Can the cache of CI machines save us from everytime downloading it.

**Comment by jiwaszki** [comment]
> In any case earlier or later, we will have to add this dependency for PyTorch Frontend validation.

Fair point, I agree with requirement addition.

> Can the cache of CI machines save us from everytime downloading it?

It will be the best case scenario, this should be clarified with CIs.

**Comment by Apollo9999** [comment]
inp1 = inp + a  # create a separate variable for updated value
inp.data = inp1.data # update the value without touching the graph

loss = torch.tensor(5.0) - inp1 # use updated value which has gradient

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.
